staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_patch.go

```diff
@@ -0,0 +1,63 @@
+package cacher
+
+import (
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+)
+
+type clusterNameAndNsGetter interface {
+	GetClusterName() string
+	GetNamespace() string
+}
+
+// recalculateWatchCacheKey adjusts the key under which the given object will be stored in the watch cache.
+//
+// resources in the db are stored without the clusterName, since the reflector used by the cache uses the logicalcluster.Wildcard
```

Comment:
This is a touch hard to follow - all objects get their clusterName assigned by the storage layer, regardless of wildcard or not. Is the cache pre-calculating the key in a non-cluster-aware manner, and that's why we are updating a previous key here, as opposed to just generating it correctly in the first place?